data_structures/stack_using_two_queues.cpp

```diff
@@ -0,0 +1,240 @@
+     */
+        bool empty() { return (q1.size() == 0); }
+    };
+
+} // namespace
+
+/**
+ * class Book with book name and it's price
+ */
+class Book
+{
+private:
+    std::string book_name{std::string()};
+    double book_price{double()};
+
+public:
+    /**
+    * default constructor
+    */
+    Book()
+    {
+    }
+
+    /**
+    * parameterized constructor
+    */
+    Book(std::string book_name, double book_price)
+        : book_name{std::move(book_name)}, book_price{std::move(book_price)}
+    {
+    }
+
+    /**
+    * Returns name of the book
+    */
+    std::string get_book_name() const
+    {
+        return this->book_name;
+    }
+
+    /**
+    * Returns price of the book
+    */
+    double get_book_price() const
+    {
+        return this->book_price;
+    }
+
+    /**
+    * comparison of two books
+    */
+    bool operator==(const Book &b) const
+    {
+        return ((b.get_book_name() == this->book_name) and (b.get_book_price() == this->book_price));
+    }
+};
+
+/**
+ * overloading operator<< to print book details
+ */
+std::ostream &operator<<(std::ostream &os, const Book &book)
+{
+    os << book.get_book_name() << ", " << book.get_book_price() << "\n";
+    return os;
+}
+
+/**
+ * utility function to print stack
+ */
+template <typename T>
+void print(Stack<T> s)
+{
+    while (!s.empty())
+    {
+        std::cout << s.top() << " ";
+        s.pop();
+    }
+    std::cout << "\n";
+}
+
+/**
+ * Testing function
+ */
+void test()
+{
+    // stack of integers
+    Stack<int> s;
+
+    // push
+    for (int i = 0; i < 10; i++)
+    {
+        s.push(i);
+    }
+
+    // iterating over stack
+    print(s);
+
+    assert(s.top() == 9);
+    std::cout << s.top() << "\n";
+
+    // size of stack
+    assert(s.size() == 10);
+    std::cout << "Stack contains " << s.size() << " elements\n";
+
+    // emptying the stack
+    while (!s.empty())
+    {
+        s.pop();
+    }
+
+    assert(s.empty() == true);
+    assert(s.top() == int());
+    std::cout << "Stack is empty: " << s.empty() << "\n";
+    std::cout << "Topmost element is: " << s.top() << "\n";
+
+    // stack of books
+    Stack<Book> books;
+
+    books.push(Book("The Philosophy of Prayer", 1122.99));
+    books.push(Book("Mahatma Gandhi Autobiography", 114.98));
+    books.push(Book("Wings of Fire", 349.86));
+    books.push(Book("My Gita", 209.95));
+    books.push(Book("The Blue Umbrella", 61.32));
+
+    print(books);
+
+    assert(books.top() == Book("The Blue Umbrella", 61.32));
+    std::cout << books.top() << "\n";
+
+    // size of stack
+    assert(books.size() == 5);
+    std::cout << "Stack contains " << books.size() << " books\n";
+
+    // emptying the stack
+    while (!books.empty())
+    {
+        books.pop();
+    }
+
+    assert(books.empty() == true);
+    assert(books.top() == Book());
+    std::cout << "Stack is empty: " << books.empty() << "\n";
+    std::cout << "Topmost element is: " << books.top() << "\n";
+}
+
+int main()
```

Comment:
Please add a comment structure like:
```suggestion
/**
 * Main function
 */
int main()
```